File: Utils/StringToURLTransformer.m
//
//  StringToURLTransformer.m
//  Cog
//
//  Created by Vincent Spader on 2/17/08.
//  Copyright 2008 __MyCompanyName__. All rights reserved.
//

#import "StringToURLTransformer.h"

@implementation StringToURLTransformer

+ (Class)transformedValueClass { return [NSURL class]; }
+ (BOOL)allowsReverseTransformation { return YES; }

// Convert from string to NSURL
- (id)transformedValue:(id)value {
    if (value == nil) return nil;

    return [NSURL URLWithString:value];
}

// Convert from NSURL to string
- (id)reverseTransformedValue:(id)value {
    if (value == nil) return nil;
    
    return [value absoluteString];
}
@end

File: Transformers/StringToURLTransformer.swift
import Foundation

@objc(StringToURLTransformer)
class StringToURLTransformer: ValueTransformer {
    override class func transformedValueClass() -> AnyClass {
        return NSURL.self
    }
    
    override class func allowsReverseTransformation() -> Bool {
        return true
    }
    
    override func transformedValue(_ value: Any?) -> Any? {
        guard let value = value else {
            return nil
        }
        
        if let string = value as? String {
            return NSURL(string: string)
        }
        
        return nil
    }
    
    override func reverseTransformedValue(_ value: Any?) -> Any? {
        guard let value = value else {
            return nil
        }
        
        if let url = value as? NSURL {
            return url.absoluteString
        }
        
        return nil
    }
}
